jobsdb/jobsdb.go

```diff
@@ -1562,14 +1564,39 @@ func (jd *HandleT) setSequenceNumber(l lock.DSListLockToken, newDSIdx string) da
 	jd.assert(len(dList) == len(dRangeList)+1 || len(dList) == len(dRangeList)+2, fmt.Sprintf("len(dList):%d != len(dRangeList):%d (+1 || +2)", len(dList), len(dRangeList)))
 
 	// Now set the min JobID for the new DS just added to be 1 more than previous max
+	var newDSMin int64
 	if len(dRangeList) > 0 {
-		newDSMin := dRangeList[len(dRangeList)-1].maxJobID
-		// jd.assert(newDSMin > 0, fmt.Sprintf("newDSMin:%d <= 0", newDSMin))
-		sqlStatement := fmt.Sprintf(`SELECT setval(pg_get_serial_sequence('"%s_jobs_%s"', 'job_id'), %d)`,
-			jd.tablePrefix, newDSIdx, newDSMin)
-		_, err := jd.dbHandle.Exec(sqlStatement)
-		jd.assertError(err)
+		newDSMin = dRangeList[len(dRangeList)-1].maxJobID + 1
+	} else {
+		newDSMin = 1
 	}
+	tableName := jd.tablePrefix + `_jobs_` + newDSIdx
+	sequenceName := tableName + `_job_id_seq`
+	// drop sequence
+	sqlStatement := fmt.Sprintf(`DROP SEQUENCE IF EXISTS "%s"`, sequenceName)
```

Comment:
According to the [documentation ](https://www.postgresql.org/docs/8.1/datatype.html#DATATYPE-SERIAL) it might be easier to create first the sequence and then the table, but what do we really gain by doing this? 

The `BIGSERIAL` data type is practically a `BIGINT` bundled with a sequence.

Or, why aren't we using the identity column syntax which supports sequence initialization as well? e.g.

```
job_id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY (START WITH 1000)
```

What about the job_status table? We have the same BIGSERIAL datatype there as well.

As a sidenote, I would find it appropriate to perform all operations within `createDS` as part of a single transaction

Reply:
> According to the [documentation ](https://www.postgresql.org/docs/8.1/datatype.html#DATATYPE-SERIAL)it might be easier to create first the sequence and then the table,

Indeed, this is what the migrator does. Over here I did it in this order just so there's least amount of discrepancy from the original code.
As for why `BIGINT`, iirc there was some issue with -ve job_ids with a `BIGSERIAL` column(can't be negative). which is why we resorted to BIGINT. The status tables won't have -ve ids, so `BIGSERIAL` suffices.
> appropriate to perform all operations within createDS as part of a single transaction

agreed